spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationExecution.scala

```diff
@@ -0,0 +1,203 @@
+    } else {
+      dfReaderConfigured
+    }).load(s"$path/*")
+    ensureSplittable(dfWithSchema, path, schema)
+  }
+
+  private def ensureSplittable(df: DataFrame, path: String, schema: StructType)
+                              (implicit spark: SparkSession, fsUtils: FileSystemVersionUtils) = {
+    if (fsUtils.isNonSplittable(path)) {
+      convertToSplittable(df, schema)
+    } else {
+      df
+    }
+  }
+
+  private def convertToSplittable(df: DataFrame, schema: StructType)
+                                 (implicit spark: SparkSession, fsUtils: FileSystemVersionUtils) = {
+    log.warn("Dataset is stored in a non-splittable format. This can have a severe performance impact.")
+
+    val tempParquetDir = s"/tmp/nonsplittable-to-parquet-${UUID.randomUUID()}"
+    log.warn(s"Converting to Parquet in temporary dir: $tempParquetDir")
+
+    // Handle renaming of source columns in case there are columns
+    // that will break because of issues in column names like spaces
+    df.select(schema.fields.map { field: StructField =>
+      renameSourceColumn(df, field)
+    }: _*).write.parquet(tempParquetDir)
+
+    fsUtils.deleteOnExit(tempParquetDir)
+    // Reload from temp parquet and reverse column renaming above
+    val dfTmp = spark.read.parquet(tempParquetDir)
+    dfTmp.select(schema.fields.map { field: StructField =>
+      reverseRenameSourceColumn(dfTmp, field)
+    }: _*)
+  }
+
+  private def renameSourceColumn(df: DataFrame, field: StructField): Column = {
+    if (field.metadata.contains(MetadataKeys.SourceColumn)) {
+      val sourceColumnName = field.metadata.getString(MetadataKeys.SourceColumn)
+      log.info(s"schema field : ${field.name} : rename : $sourceColumnName")
+      df.col(sourceColumnName).as(field.name, field.metadata)
+    } else {
+      df.col(field.name)
+    }
+  }
+
+  private def reverseRenameSourceColumn(df: DataFrame, field: StructField): Column = {
+    if (field.metadata.contains(MetadataKeys.SourceColumn)) {
+      val sourceColumnName = field.metadata.getString(MetadataKeys.SourceColumn)
+      log.info(s"schema field : $sourceColumnName : reverse rename : ${field.name}")
+      df.col(field.name).as(sourceColumnName)
+    } else {
+      df.col(field.name)
+    }
+  }
+
+  protected def standardize(dfAll: DataFrame, schema: StructType, cmd: StandardizationCmdConfig)
+                           (implicit spark: SparkSession, udfLib: UDFLibrary): DataFrame = {
+    //scalastyle:on parameter.number
+    val recordIdGenerationStrategy = getRecordIdGenerationStrategyFromConfig(conf)
+
+    try {
+      handleControlInfoValidation()
+      StandardizationInterpreter.standardize(dfAll, schema, cmd.rawFormat,
+        cmd.failOnInputNotPerSchema, recordIdGenerationStrategy)
+    } catch {
+      case e@ValidationException(msg, errors) =>
+        AtumImplicits.SparkSessionWrapper(spark).setControlMeasurementError("Schema Validation", s"$msg\nDetails: ${
+          errors.mkString("\n")
+        }", "")
```

Comment:
This is really hard to read. Could we change it a bit? Maybe extract the strings?